node-notes-cli/app.js

```diff
@@ -0,0 +1,100 @@
+import { readFile, writeFile } from 'node:fs/promises';
+
+const dataPath = 'data.json';
+
+// read the notes in data.json
+async function readNotes() {
+  try {
+    const data = await readFile(dataPath, 'utf8');
+    const notes = JSON.parse(data).notes;
+    Object.keys(notes).forEach((id) => { // Object.keys takes the notes and puts them into an array, then utilized forEach to go through every element in the array and log the id and notes
+      console.log(`${id}: ${notes[id]}`);
+    });
+  } catch (err) {
+    console.error(err.message);
+    process.exit(1);
+  }
+}
+
+// create new notes in data.json
+async function createNotes() {
+  try {
+    const data = await readFile(dataPath, 'utf8');
+    const idNotes = JSON.parse(data);
+    const newId = idNotes.nextId.toString();
+    const newNote = process.argv[3];
+    idNotes.notes[newId] = newNote;
+    const newData = JSON.stringify(idNotes, null, 2);
+    if (process.argv[3] === undefined) { // error handling if the 4th argument is undefined
```

Comment:
Should you do this check up at the top of the function, before you do any processing?

You do this in several other places. Fix them, as well.